### DISCUSSION TITLE
Build a Mac agent

### DISCUSSION THREAD

**Discussion by trhumphries** [actions/virtual-environments#8203] [Installed software]
How do I build a Mac agent? I have the Mac to build it on is there a script that I can run on the Mac to install everything?

**Comment by vpolikarpov-akvelon** [comment]
Hey @trhumphries. There is a bunch of packer templates that describes mac OS runner images: https://github.com/actions/runner-images/tree/main/images/macos/templates

Currently we have two types of templates. The old ones (without `anka` suffix) use `vsphere-clone` builder so you would need VMware vSphere environment with some base image in it. The new ones use `veertu-anka-vm-clone` builder so you would have to use Veertu's Anka environment. You my also use any other environment that can handle mac OS image and supported by packer, but you would have to adjust templates yourself.

There are no publicly available manuals of how to use these templates, but here are some links that may help you establishing build process:
- Definition for Azure pipeline that builds image: https://github.com/actions/runner-images/blob/main/images.CI/macos/azure-pipelines/image-generation.yml
- Packer tutorials: https://developer.hashicorp.com/packer/tutorials
- VMware vSphere Clone Builder docs: https://developer.hashicorp.com/packer/plugins/builders/vsphere/vsphere-clone
- Anka VM Clone Builder docs: https://developer.hashicorp.com/packer/plugins/builders/anka/vm-clone

Finally, if you just want to install runner on your bare-metal machine, please follow instructions in this repo: https://github.com/actions/runner
